backend/dal/Migrations/v01.07.03/Up/PostUp/Adminstrative_Areas_And_Addresses.sql

```diff
@@ -0,0 +1,146 @@
+-- Update addresses table first to correct locations which contain District of, City of, or The Corporation of the District of
+PRINT 'Updating Addresses table'
+
+UPDATE dbo.[Addresses]
+SET AdministrativeArea = REPLACE(AdministrativeArea, 'District of ', '')
+where AdministrativeArea like 'District of %';
+
+UPDATE dbo.[Addresses]
+SET AdministrativeArea = REPLACE(AdministrativeArea, 'City of ', '')
+where AdministrativeArea like 'City of %';
+
+UPDATE dbo.[Addresses]
+SET AdministrativeArea = REPLACE(AdministrativeArea, 'Town of ', '')
+where AdministrativeArea like 'Town of %';
+
+UPDATE dbo.[Addresses]
+SET AdministrativeArea = REPLACE(AdministrativeArea, 'Village of ', '')
+where AdministrativeArea like 'Village of %';
+
+UPDATE dbo.[Addresses]
+SET AdministrativeArea = REPLACE(AdministrativeArea, 'The Corporation of the District of ', '')
+where AdministrativeArea like 'The Corporation of the District of %';
+
+-- Do the "one off" updates to simplify the name where there is only once instance of each
+UPDATE [dbo].[Addresses]
+SET AdministrativeArea = 'Spallumcheen' 
+WHERE AdministrativeArea = 'The Corporation of the Township of Spallumcheen';
+
+PRINT 'Updating AdministrativeAreas table'
+
+-- Then update AdministrativeAreas table to fix the spelling of the Abbreviation field in order to get correct join on the Name field when doing a batch update later on
+UPDATE [dbo].[AdministrativeAreas]
+SET Abbreviation = 'Valemount' 
+WHERE Name = 'Village of Valemount' AND Abbreviation = 'Valemont';
+
+UPDATE [dbo].[AdministrativeAreas]
+SET Abbreviation = 'Fort St. John' 
+WHERE Name = 'City of Fort St John' AND Abbreviation = 'Fort St John';
+
+UPDATE [dbo].[AdministrativeAreas]
+SET Abbreviation = 'Fort St. James' 
+WHERE Name = 'District of Fort St James' AND Abbreviation = 'Fort St James';
+
+UPDATE [dbo].[AdministrativeAreas]
+SET Abbreviation = 'North Vancouver' 
+WHERE Name = 'City of North Vancouver' AND Abbreviation = 'North Vancouver - City';
+
+UPDATE [dbo].[AdministrativeAreas]
+SET Abbreviation = 'Langley' 
+WHERE Name = 'City of Langley' AND Abbreviation = 'Langley - City';
+
+-- Do the "one off" updates to simplify the name where there is only once instance of each
+UPDATE [dbo].[AdministrativeAreas]
+SET Name = 'Spallumcheen' 
+WHERE Name = 'The Corporation of the Township of Spallumcheen';
+
+UPDATE [dbo].[AdministrativeAreas]
```

Comment:
Why can't you have a single statement for all of these `update dbo.administrativeareas set name = REPLACE(name, 'District of ', '') where name like 'District of %'`?

Reply:
Because the Update statement
`UPDATE t1
SET t1.GroupName = t2.GroupName
FROM AdministrativeAreas t1
JOIN (SELECT * from AdministrativeAreas
		WHERE isnumeric(Abbreviation) = 0
		AND Abbreviation <> Name
		AND Abbreviation in (
			SELECT b.Abbreviation 
			FROM dbo.AdministrativeAreas a
			INNER JOIN AdministrativeAreas b
			ON a.Name = B.Abbreviation
		)
	) t2
ON t1.Name = t2.Abbreviation;`

 which occurs after these "one-off" updates will update the locations based on the join where the name is equal to the abbreviation since most of these one off updates would not be included in that inner join because there was only one instance of "The Corporation of the Township of Spallumcheen" and no "Spallumcheen". If I had updated the records such as "District of 100 Mile House" to just "100 Mile House" then that update script would not have matched.